tests/components/home_connect/test_init.py

```diff
@@ -294,3 +306,63 @@ async def test_services_exception(
 
     with pytest.raises(ValueError):
         await hass.services.async_call(**service_call)
+
+
+async def test_entity_migration(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    config_entry_v1_1: MockConfigEntry,
+    appliance: Mock,
+) -> None:
+    """Test entity migration."""
+
+    config_entry_v1_1.add_to_hass(hass)
+
+    device_entry = device_registry.async_get_or_create(
+        config_entry_id=config_entry_v1_1.entry_id,
+        identifiers={(DOMAIN, appliance.haId)},
+    )
+
+    test_entities = [
+        (
+            SENSOR_DOMAIN,
+            "Operation State",
+            BSH_OPERATION_STATE,
+        ),
+        (
+            SWITCH_DOMAIN,
+            "ChildLock",
+            BSH_CHILD_LOCK_STATE,
+        ),
+        (
+            SWITCH_DOMAIN,
+            "Power",
+            BSH_POWER_STATE,
+        ),
+        (
+            BINARY_SENSOR_DOMAIN,
+            "Remote Start",
+            BSH_REMOTE_START_ALLOWANCE_STATE,
+        ),
+        (
+            LIGHT_DOMAIN,
+            "Light",
+            COOKING_LIGHTING,
+        ),
+    ]
+
+    for domain, old_unique_id_suffix, _ in test_entities:
+        entity_registry.async_get_or_create(
+            domain,
+            DOMAIN,
+            f"{appliance.haId}-{old_unique_id_suffix}",
+            device_id=device_entry.id,
+            config_entry=config_entry_v1_1,
+        )
+
+    assert await config_entry_v1_1.async_migrate(hass)
+    for domain, _, expected_unique_id_suffix in test_entities:
+        assert entity_registry.async_get_entity_id(
+            domain, DOMAIN, f"{appliance.haId}-{expected_unique_id_suffix}"
+        )
```

Comment:
We should not call `async_migrate` manually, but rather have HA setup the integration correctly. We should also assert if the minor version got updated